.github/workflows/drepecated-functions-checker.yml

```diff
@@ -0,0 +1,67 @@
+name: "Check Deprecations"
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled, unlabeled]
+    # Runs on PRs to main
+    branches:
+      - main
+
+jobs:
+  deprecations-check:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0 # by default the checkout action doesn't checkout all branches
+      - name: Set up Go
+        uses: actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9 #v4
+        with:
+          go-version-file: ./.go-version
+          cache: true
+      - name: Check deprecations for files in diff
+        run: |  
+          # Staticcheck uses static analysis to finds bugs and performance issues, offers simplifications, 
+          # and enforces style rules.
+          # Here, it is used to check if a deprecated function, variable, constant or field is used.
+
+          echo "Installing staticcheck"
+          go install honnef.co/go/tools/cmd/staticcheck@2023.1.2 #v0.4.2
```

Comment:
I think even if this dependency is added to tools.go, we still have to install it in the script.

Reply:
I agree with @miagilepner it would be consistent to have the tools collected in a single place.

If we move the dependency as suggested, you could add a step to the `deprecated-functions-checker.yml` after the `setup go` step, to `install required tools` (which could just run the `go generate -tags tools tools/tools.go` mentioned in `tools.go`).

Then you can remove the `installing/go install` sections from the bash script.